src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -1174,214 +1193,255 @@ def chat(self, prompt, temperature=0.2, tools=None, output_json=None, output_pyd
+                                messages.append({"role": "user", "content": "Now regenerate your response using the reflection you made"})
+                                # For custom LLMs during reflection, always use non-streaming to ensure complete responses
+                                use_stream = self.stream if not self._using_custom_llm else False
+                                response = self._chat_completion(messages, temperature=temperature, tools=None, stream=use_stream)
+                                response_text = response.choices[0].message.content.strip()
+                                reflection_count += 1
+                                continue  # Continue the loop for more reflections
+
+                        except Exception as e:
+                                display_error(f"Error in parsing self-reflection json {e}. Retrying", console=self.console)
+                                logging.error("Reflection parsing failed.", exc_info=True)
+                                messages.append({"role": "assistant", "content": f"Self Reflection failed."})
```

Ruff (0.11.9)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)